static/js/components/Editor_test.js

```diff
@@ -0,0 +1,351 @@
+import { mount } from "enzyme"
+import { assert } from "chai"
+import { EditorView } from "prosemirror-view"
+import { schema, defaultMarkdownSerializer } from "prosemirror-markdown"
+import { toggleMark } from "prosemirror-commands"
+import * as pmMarkdown from "prosemirror-markdown"
+import * as pmCommands from "prosemirror-commands"
+
+import AddLinkMenu from "./AddLinkMenu"
+
+import { actions } from "../actions"
+import { Editor, newLinkForm, menuBarManifest } from "./Editor"
+import * as utilFuncs from "../lib/util"
+
+describe("Editor component", () => {
+  let sandbox, dispatch, onChange, makeUUIDStub, focusStub
+
+  const fakeUUID = "uuid wow hey great"
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    dispatch = sandbox.stub()
+    onChange = sandbox.stub()
+    makeUUIDStub = sandbox.stub(utilFuncs, "makeUUID").returns(fakeUUID)
+    focusStub = sandbox.stub(EditorView.prototype, "focus")
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  const renderEditor = (props = {}) =>
+    mount(<Editor dispatch={dispatch} onChange={onChange} {...props} />)
+
+  const executeTestTransaction = wrapper =>
+    toggleMark(schema.marks.strong)(
+      wrapper.instance().view.state,
+      wrapper.instance().view.dispatch,
+      wrapper.instance().view
+    )
+
+  const stubToggleMark = () => {
+    const doubleStub = sandbox.stub()
+    const toggleMarkStub = sandbox
+      .stub(pmCommands, "toggleMark")
+      .returns(doubleStub)
+    return { doubleStub, toggleMarkStub }
+  }
+
+  it("should make a new link form", () => {
+    assert.deepEqual(newLinkForm(), { url: "" })
+  })
+
+  it("should use makeUUID to set a UUID on mount", () => {
+    assert.equal(renderEditor().instance().uuid, fakeUUID)
+    assert.ok(makeUUIDStub.called)
+  })
+
+  it("should set form helpers with UUID for Link form on mount", () => {
+    const { formHelpers } = renderEditor().instance()
+    assert.isDefined(formHelpers.getForm)
+    assert.isDefined(formHelpers.formBeginEdit)
+    assert.isDefined(formHelpers.formEndEdit)
+    assert.isDefined(formHelpers.formUpdate)
+    assert.isDefined(formHelpers.formValidate)
+    const action = formHelpers.formBeginEdit()
+    assert.equal(action.payload.formKey, fakeUUID)
+  })
+
+  it("should create an EditorState and stick it in this.state", () => {
+    const wrapper = renderEditor()
+    assert.isDefined(wrapper.state().editorState)
+  })
+
+  it("should store link menu open state in this.state", () => {
+    const wrapper = renderEditor()
+    assert.isFalse(wrapper.state().showLinkMenu)
+  })
+
+  it("should put an EditorView on this.view", () => {
+    const wrapper = renderEditor()
+    assert.isDefined(wrapper.instance().view)
+  })
+
+  it("should use the initialValue prop, if passed", () => {})
+
+  //
+  ;[true, false].forEach(autoFocus => {
+    it(`${
+      autoFocus ? "should" : "shoudn't"
+    } call .focus on view if autoFocus = ${String(autoFocus)}`, () => {
```

Comment:
I don't know if I like this or if anyone else likes this, but I created this silly helper function '[shouldIf](https://github.com/mitodl/open-discussions/blob/master/static/js/lib/test_utils.js#L24)' since I was doing this all the time. This could be written as: 

```javascript
`${shouldIf(autoFocus)} call .focus on view if autoFocus = ${String(autoFocus)}`
```

You can take or leave